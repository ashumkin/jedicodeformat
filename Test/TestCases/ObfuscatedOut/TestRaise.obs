unit testraise;interface implementation uses sysutils;function test1:tdatetime;var
datestr:string;begin datestr:='Fooo';result:=0.0;try result:=strtodate(datestr);except
on econverterror do else raise;end;end;function test1_2:tdatetime;var datestr:string
;begin datestr:='Fooo';result:=0.0;try result:=strtodate(datestr);except on econverterror
do ;on ematherror do else raise;end;end;function test2:tdatetime;var datestr:string
;begin datestr:='Fooo';result:=now;try result:=strtodate(datestr);except on econverterror
do if datestr=''then raise else end;end;function test3:tdatetime;var datestr:string
;begin datestr:='Fooo';result:=now;try result:=strtodate(datestr);except on econverterror
do if datestr=''then raise else raise end;end;function test4:tdatetime;var datestr
:string;begin datestr:='Fooo';result:=now;try result:=strtodate(datestr);except on
econverterror do if datestr=''then raise else if datestr='foo'then begin raise end
else raise end;end;procedure testif(const pi:integer);begin if pi>4 then else begin
end;end;end. 