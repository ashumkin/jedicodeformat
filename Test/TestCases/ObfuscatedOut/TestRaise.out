unit testraise;

interface

implementation

uses SysUtils;

function test1: tdatetime;
var
  datestr: string;
begin
  datestr := 'Fooo';
  Result  := 0.0;
  try
    Result := strtodate(datestr);
  except
    on econverterror do
    else
      raise;
  end;
end;

function test1_2: tdatetime;
var
  datestr: string;
begin
  datestr := 'Fooo';
  Result  := 0.0;
  try
    Result := strtodate(datestr);
  except
    on econverterror do ;
    on ematherror do
    else
      raise;
  end;
end;

function test2: tdatetime;
var
  datestr: string;
begin
  datestr := 'Fooo';
  Result  := now;
  try
    Result := strtodate(datestr);
  except
    on econverterror do
      if datestr = '' then
        raise
      else
  end;
end;

function test3: tdatetime;
var
  datestr: string;
begin
  datestr := 'Fooo';
  Result  := now;
  try
    Result := strtodate(datestr);
  except
    on econverterror do
      if datestr = '' then
        raise
      else
        raise
  end;
end;

function test4: tdatetime;
var
  datestr: string;
begin
  datestr := 'Fooo';
  Result  := now;
  try
    Result := strtodate(datestr);
  except
    on
      econverterror do
      if datestr = '' then
        raise
      else if datestr = 'foo' then
      begin
        raise;
      end
      else
        raise
  end;
end;

procedure testif(const pi: integer);
begin
  if pi > 4 then
  else
  begin
  end;
end;

end.
 